File: transact-example.cs
using System;
using System.Text;
using System.Net;
using System.IO;

namespace CornerstoneTransactExample
{
	class Program
	{
		static void Main()
		{
			string url   = "https://api.cornerstone.cc/v1/transactions";
			string user  = "sandbox_3xSOjtxSvICXVOKYqbwI";
			string key   = "key_RdutJGqI50YIwjehGtHBOe1Uu";

			// you will probably want to create this xml
			// string using a library or class
			var postData =
				"<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
				"<request>" +
					"<amount>15</amount>" +
					"<card>" +
						"<number>4444333322221111</number>" +
						"<expmonth>12</expmonth>" +
						"<expyear>23</expyear>" +
						"<cvv>999</cvv>" +
					"</card>" +
					"<customer>" +
						"<firstname>Robert</firstname>" +
						"<lastname>Parr</lastname>" +
						"<email>robertp@example.com</email>" +
					"</customer>" +
				"</request>";
			var data = Encoding.ASCII.GetBytes(postData);

			string authInfo = user + ":" + key;
			authInfo = Convert.ToBase64String(Encoding.Default.GetBytes(authInfo));

			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
			request.Method        = "POST";
			request.ContentType   = "application/xml";
			request.Accept        = "application/xml";
			request.ContentLength = data.Length;

			request.Headers["Authorization"] = "Basic " + authInfo;

			using (var stream = request.GetRequestStream())
			{
				stream.Write(data, 0, data.Length);
			}

			HttpWebResponse response = null;

			try {
				response = (HttpWebResponse)request.GetResponse();
			} catch (WebException e) {
				if (e.Status == WebExceptionStatus.ProtocolError) {
					response = (HttpWebResponse)e.Response;
					Console.Write("Errorcode: {0}\n", (int)response.StatusCode);
				} else {
					Console.Write("Error: {0}\n", e.Status);
				}
			} finally {
				if (response != null) {
					string responseString = new StreamReader(response.GetResponseStream()).ReadToEnd();
					Console.WriteLine(responseString);
					response.Close();
				}
			}
		}
	}
}

File: transact-example.java
import java.io.*;
import java.net.*;

class CornerstoneTransactionExample {
	public static void main(String[] args) {
		String url = "https://api.cornerstone.cc/v1/transactions";
		String user = "sandbox_3xSOjtxSvICXVOKYqbwI";
		String key = "key_RdutJGqI50YIwjehGtHBOe1Uu";
		String result = "";
		
		// you will likely want to use something like jaxb or xstream
		// to create xml and possibly write it to the stream
		String request = 
		"<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
		"<request>" +
			"<amount>15</amount>" +
			"<card>" +
				"<number>4444333322221111</number>" +
				"<expmonth>12</expmonth>" +
				"<expyear>23</expyear>" +
			"</card>" +
			"<customer>" +
				"<firstname>Robert</firstname>" +
				"<lastname>Parr</lastname>" +
				"<email>robertp@example.com</email>" +
			"</customer>" +
		"</request>";

		try {				
			HttpURLConnection con = (HttpURLConnection) (new URL(url)).openConnection();
			
			byte[] message = (user+":"+key).getBytes("UTF-8");
			String encoded = javax.xml.bind.DatatypeConverter.printBase64Binary(message);
						
			con.setRequestProperty("User-Agent", user);
			con.setRequestProperty("Authorization", "Basic " + encoded);
			con.setRequestProperty("Accept", "application/xml");
			con.setRequestProperty("Content-Type", "application/xml");

			con.setRequestMethod("POST");
			con.setDoOutput(true);
			
			DataOutputStream out = new DataOutputStream(con.getOutputStream());
			out.writeBytes(request);
			out.flush();
			out.close();
			
			InputStream stream = con.getResponseCode() < 400 ? con.getInputStream() : con.getErrorStream();
						
			BufferedReader buf = new BufferedReader(new InputStreamReader(stream));
			
			String line;
			while ((line = buf.readLine()) != null) {
				result += line + "\n";
			}

			System.out.println(result);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}
